resource-managers/kubernetes/integration-tests/src/test/scala/org/apache/spark/deploy/kubernetes/integrationtest/KubernetesV2Suite.scala

```diff
@@ -135,6 +135,19 @@ private[spark] class KubernetesV2Suite(testBackend: IntegrationTestBackend)
     runSparkPiAndVerifyCompletion(SparkLauncher.NO_RESOURCE)
   }
 
+  test("Use client key and client cert file when requesting executors") {
```

Comment:
I was under the impression it _either_ needs the service account token or the client key-cert pair. It's unfortunately difficult to test just this in isolation since we fall back to using the service account token if the user does not provide other credentials.